cicecore/cicedynB/infrastructure/comm/mpi/ice_global_reductions.F90

```diff
@@ -1683,6 +1685,56 @@ function global_maxval_scalar_int (scalar, dist) &
 
  end function global_maxval_scalar_int
 
+!***********************************************************************
+
+ function global_maxval_scalar_int_nodist (scalar, communicator) &
```

Comment:
Good question.  The answer is a little complicated.  I added this because I needed it for testing.  In the bcstchk (ice_broadcast check) unit test, I broadcast data from one pe to others using the ice_broadcast (mpi or serial) infrastructure.  Then I have to validate that.  To validate it, I have each pe check it's results and then I broadcast that result to all pes, so now there is a global understanding of whether the test passed.  To do that, I need MPI when running on multiple pes but I can't have any MPI when running on a single pe testing serial.  So, that broadcast (which is really implemented as an MPI_MAX global reduction) needed to live in the serial/mpi directories rather than in the unit test because we don't have a "NO_MPI" cpp yet.  Because all of the global_reduction infrastructure requires a "distribution" (but the ice_broadcast does not and the bcstchk has to grid/decomposition), I decided to implement a global reduction with just a communicator, basically as a simple layer on MPI.

If this is undesirable, there are a few other things I could do
- I could leverage the existing global reduction infrastructure, but this would mean the bcstchk unit tester would have to initialize a grid/decomposition even though it's not needed for the unit test.
- We could add a cpp something like "NO_MPI" and I could then leverage that in the bcstchk itself and implement the MPI global reduction there.  

I prefer not to do the first option.  The second option is something that maybe we want to do eventually anyway, so I could start to implement it now.   All the mpi and "no_mpi" code is segregated into the serial and mpi directories right now.  Ultimately, we may want to merge those directories in which case we'd need a cpp.